tools/setup/emoji/generate_emoji_listing

```diff
@@ -0,0 +1,219 @@
+                    <th class="zulip-emoji-names">Zulip names</th>
+                    <th class="iamcal-emoji-names">Iamcal names</th>
+                    <th class="gemoji-emoji-names">Gemoji names</th>
+                    <th class="unicode-name">Unicode name</th>
+                </tr>
+            </thead>
+            <tbody>
+                %(tbody)s
+            </tbody>
+        </table>
+    </body>
+</html>
+"""
+
+TABLE_CSS = """
+    .emoji {
+        height: 35px;
+        width: 35px;
+        position: relative;
+        margin-top: -7px;
+        vertical-align: middle;
+        top: 3px;
+    }
+
+    .emoji-images {
+        width: 200px;
+    }
+
+    table, td, th {
+        border: 1px solid black;
+        border-collapse: collapse;
+    }
+
+    td, th {
+        height: 40px;
+        text-align: center;
+    }
+
+    .google {
+        background-image: url('sheet_google_32.png') !important;
+    }
+
+    .apple {
+        background-image: url('sheet_apple_32.png') !important;
+    }
+
+    .emojione {
+        background-image: url('sheet_emojione_32.png') !important;
+    }
+
+    .twitter {
+        background-image: url('sheet_twitter_32.png') !important;
+    }
+"""
+
+SORTED_CATEGORIES = [
+    'People',
+    'Nature',
+    'Foods',
+    'Activity',
+    'Places',
+    'Objects',
+    'Symbols',
+    'Flags',
+    'Skin Tones',
+]
+
+emoji_code_to_zulip_names = {}      # type: Dict[str, str]
+emoji_code_to_iamcal_names = {}     # type: Dict[str, str]
+emoji_code_to_gemoji_names = {}     # type: Dict[str, str]
+emoji_collection = {category: [] for category in SORTED_CATEGORIES}     # type: Dict[str, List[Dict[str, Any]]]
+
+def generate_emoji_code_to_emoji_names_maps():
+    # type: () -> None
+    # Prepare gemoji names map.
+    reverse_unified_reactions_map = {}  # type: Dict[str, List[str]]
+    for name in UNIFIED_REACTIONS_MAP:
+        emoji_code = UNIFIED_REACTIONS_MAP[name]
+        if emoji_code in reverse_unified_reactions_map:
+            reverse_unified_reactions_map[emoji_code].append(name)
+        else:
+            reverse_unified_reactions_map[emoji_code] = [name, ]
+
+    for emoji_code in reverse_unified_reactions_map:
+        emoji_code_to_gemoji_names[emoji_code] = ", ".join(reverse_unified_reactions_map[emoji_code])
+
+    # Prepare iamcal names map.
+    for emoji_dict in EMOJI_DATA:
+        emoji_code = emoji_dict["unified"].lower()
+        emoji_code_to_iamcal_names[emoji_code] = ", ".join(emoji_dict["short_names"])
+
+    # Prepare zulip names map.
+    for emoji_code in ZULIP_EMOJI_MAP:
+        canonical_name = ZULIP_EMOJI_MAP[emoji_code]["canonical_name"]
+        aliases = ZULIP_EMOJI_MAP[emoji_code]["aliases"]
+        names = [canonical_name, ]
+        names.extend(aliases)
+        emoji_code_to_zulip_names[emoji_code] = ", ".join(names)
+
+def get_sorting_info(category, sort_order):
+    # type: (str, int) -> str
+    return ", ".join([category, str(sort_order)])
```

Comment:
I think no comma here. "People 28" is easier to search for than "People, 28".

Reply:
Done!